File: leetcode/editor/en/[2038]Remove Colored Pieces if Both Neighbors are the Same Color.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public boolean winnerOfGame(String colors) {
        int countA = 0;
        int countB = 0;
        char[] c = colors.toCharArray();

        for (int i = 2; i < c.length; i++) {
            if (c[i] == 'A' && c[i - 1] == 'A' && c[i - 2] == 'A') {
                countA++;
            }
            
            if (c[i] == 'B' && c[i - 1] == 'B' && c[i - 2] == 'B') {
                countB++;
            }
        }

        return countA > countB;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2038]Remove Colored Pieces if Both Neighbors are the Same Color.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} colors
 * @return {boolean}
 */
var winnerOfGame = function (colors) {
    let countA = 0;
    let countB = 0;

    for (let i = 2; i < colors.length; i++) {
        if (colors[i] === 'A' && colors[i - 1] === 'A' && colors[i - 2] === 'A') {
            countA++;
        }

        if (colors[i] === 'B' && colors[i - 1] === 'B' && colors[i - 2] === 'B') {
            countB++;
        }
    }

    return countA > countB;
};
//leetcode submit region end(Prohibit modification and deletion)
